manage/heroes/new.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+require '../layout/header.php';
+
+use Model\hero;
+use Service\Container;
+use Model\AbstractShip; // get the teams
+
+$teams = AbstractShip::getTeams();
+$errors = [];
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $team = $_POST['team'];
+    if (empty($team)) {
+        $errors = 'All ships must have an allegiance';
+    } elseif (!in_array($team, $teams)) {
+        $errors[] = 'Select a valid team';
+    }
```

Comment:
Check for a valid team (see note in Hero.php)

Reply:
getValidTeams() created and implemented